Add Gravatar to logged-in verbum comments

johngodley opened the pull request:
In #42458 we added Gravatar profiles to logged out comments. This PR adds them to logged in comments.

The change affects the comment settings button shown at the bottom of Jetpack comments when the user is logged in:

![image](https://github.com/user-attachments/assets/0bdbfea8-e587-4bbf-8f2c-78d08aaec593)

There is no design change to this button, but it now functions in a different way and rather than behave as a single button it is split between direct access to the Quick Editor, and access to the comment settings.

Clicking the Gravatar image will open up the Gravatar Quick Editor, allowing the user to update their avatar. Any updated avatar will be reflected in the comment image after saving.

Clicking the cog icon opens up the comment settings, as before, but with the addition of an 'Edit Gravatar' button:

![image](https://github.com/user-attachments/assets/8f078001-8369-4151-92d9-f3cb5fa23a26)

The quick editor is dynamically loaded and so doesn't impact the size of the initial load when the user is logged in.

Some code has been refactor and this has affected the size of the load when not logged in. However, this is a minor shift in dynamic bundle size from one file to another (and only loaded when the user shows the comment form).

## Proposed changes:
* Modifies the logged in comment settings button, splitting it in 2 (although it looks the same)
* Provides direct access to the Gravatar Quick Editor by clicking the comment setting Gravatar image
* Adds an 'Edit Gravatar' button. when the comment settings are expanded

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [X] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

It will be useful to perform these tests in 2 windows to compare the appearance before and after.

Run `bin/jetpack-downloader test jetpack-mu-wpcom-plugin update/gravatar-verbum-logged-in` to copy the PR to your sandbox and make sure `jetpack.wordpress.com` is sandboxed

* Load a post as a logged-in user. Confirm that the comment settings area still shows a Gravatar and a settings button as before, and the same as the above screenshot
* Click the image and confirm the Gravatar Quick Editor is opened
* Change your avatar and wait for it to be saved. The image in the comment form should flash and be updated
* Close the quick editor
* Click the settings cog icon and confirm the appearance remains unchanged but with a new 'Edit Gravatar' button
* Click this and perform a similar avatar change
* Tabbing through the comment form should show a blue outline around each button individually
![image](https://github.com/user-attachments/assets/e135be20-6617-48a1-ac52-d73292feee21) 
![image](https://github.com/user-attachments/assets/8d7b7228-5521-48e2-851f-dc296546816a)

johngodley committed:
Move quick editor to a seperate file

johngodley committed:
Modify logged in comment button to show Gravatar

The button opens the quick editor when pressed, but otherwise behaves the same as before

johngodley committed:
Add changelog

johngodley committed:
Remove extra import

aaronfc reviewed (dismissed):
Looks good and works as described! 🚀 

I have some personal feedback though: the "Edit Gravatar" link looks a bit out of place.
I would say it looks even like a link with a broken style to me (as a user).

Would it be possible to add that option inside the form? But honestly I did this mockup and doesn't seem much better... and it might be a bit "too far away" from the Gravatar itself.

<img width="589" alt="image" src="https://github.com/user-attachments/assets/6757c00d-f0fe-448d-a455-5b93fdc4296d" />

johngodley commented:
> Would it be possible to add that option inside the form? But honestly I did this mockup and doesn't seem much better... and it might be a bit "too far away" from the Gravatar itself.

We talked about this in Slack and will leave the button as-is for now.

johngodley committed:
Micro adjustments to the buttons

The buttons are positioned slightly differently and need some adjustments so they look the same

johngodley commented:
I noticed there was a border around the buttons on simple sites (not using jetpack.wordpress.com), and so had to make some adjustments. Comparing it to a previous version I also noticed there were a few pixels of margin different, which I've also tried to account for.

I think this needs another quick test on both an Atomic and Simple site just to make sure everything is lined up.

jom reviewed (approved):
Looks good; works well!
